src/proc/constant_evaluator.rs

```diff
@@ -305,20 +307,31 @@ impl ConstantEvaluator<'_> {
                 let expr = Expression::Splat { size, value };
                 Ok(self.register_constant(expr, span))
             }
-            Expression::Compose {
-                ty,
-                components: ref src_components,
-            } => {
+            Expression::Compose { ty, ref components } => {
                 let dst_ty = get_dst_ty(ty)?;
 
-                let components = pattern
+                let mut flattened = [src_constant; 4]; // dummy value
+                let len = self
+                    .flatten_compose(ty, components)
+                    .zip(flattened.iter_mut())
+                    .map(|(component, elt)| *elt = component)
+                    .count();
```

Comment:
Very interesting use of iterators - haven't seen/used them this way before.

Reply:
Yeah, `zip` has the perfect "shorter of the two" behavior. And `count` forces exhaustion of the iterator. So it's cute.

In general "cute" is not a good thing - but I think it's fine when its impact is strictly local.